File: src/main/java/com/hst/sitescrapper/controller/ScrapController.java
package com.hst.sitescrapper.controller;

import com.hst.sitescrapper.model.api.ApiResponse;
import com.hst.sitescrapper.model.request.ScrapRequest;
import com.hst.sitescrapper.service.ScrapService;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("scrap")
public class ScrapController {

    private Logger log = LoggerFactory.getLogger(getClass());

    @Autowired
    private ScrapService scrapService;

    @GetMapping
    public ResponseEntity<ApiResponse> findWebSlices() {
        return new ResponseEntity<>(ApiResponse.of(100, "success", scrapService.findWebSlices()), HttpStatus.OK);
    }

    @PostMapping
    public ResponseEntity<ApiResponse> createWebSlice(@RequestBody ScrapRequest scrapRequest) {
        log.info("New scrap request {}", scrapRequest);

        scrapService.createWebSlice(scrapRequest);

        return new ResponseEntity<>(ApiResponse.of(100, "success"), HttpStatus.OK);
    }



}

File: backend/src/main/kotlin/com/hst/scrapper/scrap/ui/ScrapController.kt
package com.hst.scrapper.scrap.ui

import com.hst.scrapper.global.model.api.ApiResponse
import com.hst.scrapper.scrap.domain.application.ScrapService
import com.hst.scrapper.scrap.ui.response.ScrapResponse
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

/**
 * @author dlgusrb0808@gmail.com
 */
@RestController
@RequestMapping("/api/scraps")
class ScrapController(
    val scrapService: ScrapService
) {

    @GetMapping
    fun getScraps(@RequestParam userId: Long): ApiResponse<List<ScrapResponse>> {
        return ApiResponse.success(scrapService.getScraps(userId))
    }

}